common/app/model/Cors.scala

```diff
@@ -8,20 +8,25 @@ object Cors extends Results with implicits.Requests {
 
   private val defaultAllowHeaders = List("X-Requested-With","Origin","Accept","Content-Type")
 
-  def apply(result: Result, allowedMethods: Option[String] = None, fallbackAllowOrigin: String = "*")(implicit request: RequestHeader): Result = {
+  def apply(result: Result, allowedMethods: Option[String] = None, fallbackAllowOrigin: Option[String] = None)(implicit request: RequestHeader): Result = {
 
     val responseHeaders = (defaultAllowHeaders ++ request.headers.get("Access-Control-Request-Headers").toList) mkString ","
 
     request.headers.get("Origin") match {
       case None => result
       case Some(requestOrigin) =>
-        val allowedOrigin = ajax.corsOrigins.find(_ == requestOrigin).getOrElse(fallbackAllowOrigin)
-        val headers = allowedMethods.map("Access-Control-Allow-Methods" -> _).toList ++ List(
-          "Access-Control-Allow-Origin" -> allowedOrigin,
-          "Access-Control-Allow-Headers" -> responseHeaders,
-          "Access-Control-Allow-Credentials" -> "true")
+        val allowedOrigin = ajax.corsOrigins.find(_ == requestOrigin).orElse(fallbackAllowOrigin)
 
-        result.withHeaders(headers: _*)
+        allowedOrigin match {
+          case Some(origin) =>
+            val headers = allowedMethods.map("Access-Control-Allow-Methods" -> _).toList ++ List(
+              "Access-Control-Allow-Origin" -> origin,
+              "Access-Control-Allow-Headers" -> responseHeaders,
+              "Access-Control-Allow-Credentials" -> "true")
+
+            result.withHeaders(headers: _*)
+          case None => Forbidden("Unsupported CORS origin")
```

Comment:
I think just not returning the `Access-Control-Allow-Origin` header will produce the same behaviour in browsers, though returning a 403 might be better as we can see in the logs how often it happens?

Reply:
Ah maybe that is better. I wasn't sure what was the 'right' approach to error handling here.

Reply:
yep I'm not too sure either: the 403 is nice as we can easily see if anyone is affected.